### PR TITLE
Feature/functions record vote

### PR DESCRIPTION
**Pull request by stackswell**
# Description
---
This is a request to merge current working code for `/functions/recordVote` into the development branch.

## Changelog
- Implements vote recording within `poll` object, leveraging a map of `votedFor` values keyed by unique voter phone number.
 - Applies regex match for `voteFor` values loosely matching `shortcode` or `name` values of options.
 - Implements `totalVotes` counter to be incremented per matched vote attempt.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,7 @@ client/public/*
 .idea/**/usage.statistics.xml
 .idea/**/dictionaries
 .idea/**/shelf
+.idea/*
```
Comment (aslattery) [inline .gitignore]: @stackswell just curious, are you having to do this in other repositories too?

File: functions/recordVote/index.js [hunk 2]
```diff
@@ -1,5 +1,107 @@
...
+        Body: ‘Test two’,
+        FromCountry: ‘US’,
+        To: ‘+12172884421’,
+        MessagingServiceSid: ‘MG2ca64febf3dc8f8eb0fd10d7aa45663b’,
+        ToZip: ‘61810’,
+        NumSegments: ‘1’,
+        MessageSid: ‘SM0502c78af796157271062c0155e5f5ec’,
+        AccountSid: ‘AC0e0f269bf9ec4a806d4dbe5a1c616023’,
+        From: ‘+15203707474’,
+        ApiVersion: ‘2010-04-01’
```
Comment (aslattery) [inline functions/recordVote/index.js]: @stackswell would recommend removing this or sanitizing it, as it contains your PII.

Comment (stackswell) [reply functions/recordVote/index.js]: Really not a concern for me, pushed a commit removing all relevant data. Feel free to merge at will

Comment (aslattery) [reply functions/recordVote/index.js]: Roger that.

File: .gitignore (hunk 1 above)
Comment (stackswell) [reply .gitignore]: @aslattery The generated sub-directory tree seemingly changes every few months and the boilerplate gitignore almost never covers all generated files, so yes for sanity sake I do this on most projects

Comment (aslattery) [reply .gitignore]: Got it, I'll remember to simplify this rule in future projects.

**Pull request merged**